administrator/components/com_admin/sql/updates/postgresql/4.2.0-2022-05-15.sql

```diff
@@ -0,0 +1,63 @@
+--
+-- Create the new table for MFA
+--
+CREATE TABLE IF NOT EXISTS "#__user_mfa" (
+  "id" serial NOT NULL,
+  "user_id" bigint NOT NULL,
+  "title" varchar(255) DEFAULT '' NOT NULL,
+  "method" varchar(100) NOT NULL,
+  "default" smallint DEFAULT 0 NOT NULL,
+  "options" text NOT NULL,
+  "created_on" timestamp without time zone NOT NULL,
+  "last_used" timestamp without time zone,
+  PRIMARY KEY ("id")
+);
+
+CREATE INDEX "#__user_mfa_idx_user_id" ON "#__user_mfa" ("user_id") /** CAN FAIL **/;
+
+COMMENT ON TABLE "#__user_mfa" IS 'Multi-factor Authentication settings';
+
+--
+-- Remove obsolete postinstallation message
+--
+DELETE FROM "#__postinstall_messages" WHERE "condition_file" = 'site://plugins/twofactorauth/totp/postinstall/actions.php';
+
+--
+-- Add new MFA plugins
+--
+INSERT INTO "#__extensions" ("package_id", "name", "type", "element", "folder", "client_id", "enabled", "access", "protected", "locked", "manifest_cache", "params", "custom_data", "ordering", "state") VALUES
+(0, 'plg_multifactorauth_totp', 'plugin', 'totp', 'multifactorauth', 0, 0, 1, 0, 1, '', '', '', 1, 0),
+(0, 'plg_multifactorauth_yubikey', 'plugin', 'yubikey', 'multifactorauth', 0, 0, 1, 0, 1, '', '', '', 2, 0),
+(0, 'plg_multifactorauth_webauthn', 'plugin', 'webauthn', 'multifactorauth', 0, 0, 1, 0, 1, '', '', '', 3, 0),
+(0, 'plg_multifactorauth_email', 'plugin', 'email', 'multifactorauth', 0, 0, 1, 0, 1, '', '', '', 4, 0),
+(0, 'plg_multifactorauth_fixed', 'plugin', 'fixed', 'multifactorauth', 0, 0, 1, 0, 1, '', '', '', 5, 0);
+
+--
+-- Update MFA plugins' publish status
+--
+UPDATE "#__extensions" AS "a"
+SET "a"."enabled" = "b"."enabled"
+FROM "#__extensions"
```

Comment:
```suggestion
SET "enabled" = "b"."enabled"
FROM "#__extensions" AS "b"
```
Fix SQL error `ERROR:  missing FROM-clause entry for table "b"` in line 39.

The target columns in the "SET" clause shall not have an alias, so the `SET "a"."enabled" = "b"."enabled"` needs to be changed to `SET "enabled" = "b"."enabled"` in line 39.

When that is fixed, the next SQL error is the missing table alias "b", so it needs to add a ` AS "b"` to line 40.

Reply:
Thanks. I do not have PostgreSQL on this machine and I'm out of office until the end of the week; the changes there were made blind.

Reply:
Well I have tested that particular SQL statement here with PostgreSQL because I knew it could be critical. But I haven't done a complete test yet.

Reply:
Beyond the installation and update SQL, everything else is bog standard ANSI SQL, really. I don't expect any problems but you know what they tell about assumptions so, please, if you do have the time do test :) Thank you!